pkg/vpn/vpn.go

```diff
@@ -90,6 +93,9 @@ func getVPNAuthInfo(vpnAuth string) (vpnCliAuthInfo, error) {
 			authInfo.JoinKey = vpnKeyValue[1]
 		case "controlServerURL":
 			authInfo.ControlServerURL = vpnKeyValue[1]
+		case "extraArgs":
+			extraArgs := strings.TrimPrefix(vpnKeyValues, "extraArgs=")
+			authInfo.ExtraCLIFlags = strings.Fields(extraArgs)
```

Comment:
use of `strings.Fields` seems to imply that users won't ever need to pass args with spaces in them? For example, this doesn't seem to work right: `name=foo,joinKey=bar,extraArgs=--accept-dns=false --netfilter-mode=false --foo="bar baz"`

Reply:
Thanks for the review Brad. Yes, you are correct, all flags for `tailscale up` should be correctly covered when using `strings.Fields`. The flags that allow more than one string, e.g. `--advertise-routes` always require the strings to be comma separated: `"10.0.0.0/8,192.168.0.0/24"`.

But, I have now realized that perhaps users add a space after the comma `"10.0.0.0/8, 192.168.0.0/24"`, so maybe I should find another way. I'll try using `Split` with `--` as the separator

Reply:
After complicating the code quite a bit, I managed to cover the space use case. However, I have just realized that tailscale does not accept space in the flags values, so for example `--advertise-routes=10.0.0.0/8, 192.168.0.0/24`, fails when passed to tailscale binary. Same for other []strings flags like `advertise-tags`.  Therefore, I'll revert to the `Fields` commit which was simpler. I'll add some comments in the code so that we remember

Reply:
Yeah, I guess that we probably don't need to worry about spaces at the moment as `tailscale up` doesn't appear to have any arguments that would accept a space in their value... but it might be good to add to our test coverage and handled it properly anyway?

Reply:
You suggest to still handle the space case even if tailscale does not accept it?

Reply:
If it's not too hard to do, it would be nice to handle it properly so we don't get complaints about it later. If it's not trivial, then forget it.

Reply:
I prefer to leave it as it is. Tailscale people have just confirmed that they don't expect to change those rules: spaces are and will not be accepted